pkg/model/config/config.go

```diff
@@ -147,44 +220,84 @@ func (c Config) resourceAPIVersionCompatible(verType, version string) bool {
 	return true
 }
 
-// GVK contains information about scaffolded resources
-type GVK struct {
-	Group   string `json:"group,omitempty"`
-	Version string `json:"version,omitempty"`
-	Kind    string `json:"kind,omitempty"`
+// simplify omits some fields that can be restored at unmarshalling.
+func (c *Config) simplify() {
+	if c.IsV3() {
+		for i, r := range c.Resources {
+			// If the plural is regular, omit it.
+			if r.Plural == resource.RegularPlural(r.Kind) {
+				c.Resources[i].Plural = ""
+			}
+			// If the path is the default location, omit it.
+			if r.Path == resource.LocalPath(c.Repo, r.Group, r.Version, c.MultiGroup) {
+				c.Resources[i].Path = ""
+			}
+			// If API is empty, omit it (prevents `api: {}`).
+			if r.API != nil && r.API.IsEmpty() {
+				c.Resources[i].API = nil
+			}
+			// If Webhooks is empty, omit it (prevents `webhooks: {}`).
+			if r.Webhooks != nil && r.Webhooks.IsEmpty() {
+				c.Resources[i].Webhooks = nil
+			}
+		}
+	}
+}
 
-	// CRDVersion holds the CustomResourceDefinition API version used for the GVK.
-	CRDVersion string `json:"crdVersion,omitempty"`
-	// WebhookVersion holds the {Validating,Mutating}WebhookConfiguration API version used for the GVK.
-	WebhookVersion string `json:"webhookVersion,omitempty"`
+// restore sets the omitted values that were simplified during marshalling.
+func (c *Config) restore() {
+	if c.IsV3() {
```

Comment:
It would be better if the code is valid for any no v2 project. See: https://github.com/kubernetes-sigs/kubebuilder/pull/1696/files#diff-5e3c06a663e77e82b3da9e45c08d6b209590130fcf612ebf2af9343e1dd71f02R260-R295

Reply:
The code is valid for all projects. Project version 2 doesn't store plurals so we do not need to restore them, and the same for the rest of the fields.

Reply:
No, it is not. See that you are `c.IsV3()` which means that it will not be valid for the future versions.

Reply:
Why it would not be valid for the future versions? Should we not for v4/ for examples, `// Restore regular plural that were omitted.` and `// Restore the default location that was omitted.` as well?

Reply:
We have `if c.IsV3() {` then if is v4 should the restore works as it works for v3 or not? If not, why? 

>Project version 2 doesn't store plurals so we do not need to restore them,

I do not agree with the approach of this restore because it reduces the maintainability because instead of we are centralizing the plugin caveats in the plugin we are pushing to the model layer that is shared and used by any version. Also, it is more susceptible to errors. We should read the project file and populate the config with the info that it has or not and then, each plugin should know how to use that accordingly.

Reply:
It needs to be `!c.IsV2()`